src/main/java/org/terasology/flexiblepathfinding/PathfinderSystem.java

```diff
@@ -84,21 +78,42 @@ public int requestPath(EntityRef requester, Vector3i target, List<Vector3i> star
         return requestPath(config, callback);
     }
 
+    private void processPath(ConfigWrapper wrapper) {
+        JPSImpl jps = new JPSImpl(wrapper.config);
+        List<Vector3i> path = Lists.newArrayList();
+        try {
+            if (jps.run()) {
+                path = jps.getPath();
+            }
+        } catch (InterruptedException e) {
+            // do nothing
+        }
+        this.completePathFor(wrapper.config.requester);
+        wrapper.callback.pathReady(path, new Vector3i()); // unusued :??
+    }
+
     public int requestPath(JPSConfig config, PathfinderCallback callback) {
         if (config.requester != null && config.requester.exists()) {
             if (entitiesWithPendingTasks.contains(config.requester)) {
                 return -1;
             }
             entitiesWithPendingTasks.add(config.requester);
         }
+        Observable.just(new ConfigWrapper(callback, config))
+                .subscribeOn(GameThread.io())
```

Comment:
Why `io()` ?

Reply:
its on a blocking thread where the computation is then passed off to another threading doing the actual computation.

Reply:
`io()` is no longer used in the current state of the PR.